python/build-module.py

```diff
@@ -40,28 +40,20 @@ def build_cython_extensions() -> None:
         extra_compile_args = [
             "-O3",
         ]
-    # Relative to project root directory
-    include_dirs = {
-        "pyiceberg/",
-    }
 
-    extensions = [
```

Comment:
I've cleaned this a bit up because we don't rely on other c-bindings, so the included dirs are not required. This reduces the number of paths that we pass in (and that might be different on other platforms).